### ISSUE TITLE
reload and restart - smart insertion of new tasks

### ISSUE THREAD

**Issue by arjclark** [cylc/cylc-flow#774]
Result of an email conversation between myself, @hjoliver and @dpmatthews.

It would be useful to have cylc perform a smart insertion of new tasks on performing a reload.

Options could include:
- insert at latest cycle
- insert at earliest (in focus) cycle
- an option to specify the cycle at which to perform the insertion

There may be a challenge with determining what to do about purged tasks and whether they are "new", depending on how the suite tracks those operations across reloads.

**Comment by hjoliver** [comment]
Worth doing - some users do seem to find the lack of new task auto-insertion confusing.

**Comment by hjoliver** [comment]
Find earliest instance of an unsatisfied task that depends on a new task - insert there.

**Comment by oliver-sanders** [comment]
[meeting] A robust, reliable solution that works for all edge cases is a major challenge. A solution which fails to handle these edge cases properly risks generating a false sense of security. Once we move to a spawn-on-demmand model for the task pool, auto-insert on reload should become somewhat easier.

**Comment by oliver-sanders** [comment]
To be closed with or after #987

**Comment by oliver-sanders** [comment]
So very happy to close this, I can finally delete my failed attempts.
